### PR TITLE
sinkv2(ticdc): add sinkv2 mysql DML utils

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #5928 

### What is changed and how it works?

Add sinkv2 mysql DML utils.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by liuzix** [comment]
- Can `pkg/sqlmodel` be used?
- I suggest making an interface `dmlPreparer`. We can keep the original logic for now, and replace the implementation with one based on `pkg/sqlmodel` when it is fully tested.

File: cdc/sinkv2/eventsink/txn/mysql/dml.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+	err = db.PingContext(ctx)
+	if err != nil {
+		// close db to recycle resources
+		if closeErr := db.Close(); closeErr != nil {
+			log.Warn("close db failed", zap.Error(err))
+		}
+		return nil, cerror.ErrMySQLConnectionError.Wrap(err).GenWithStack("fail to open MySQL connection")
+	}
+	return db, nil
+}
```
Comment (liuzix) [inline cdc/sinkv2/eventsink/txn/mysql/dml.go:255]: We should have an interface `DBConnFactory`.

**Review by liuzix — APPROVED** [review]
We can factor out interfaces later.

File: cdc/sinkv2/eventsink/txn/mysql/dml.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+				query = replace
+				cacheCount = 0
+				cacheArgs = make([]interface{}, 0, len(cacheArgs))
+				last = false
+			}
+		}
+	}
+	return sqls, args
+}
+
```
Comment (nongfushanquan) [inline cdc/sinkv2/eventsink/txn/mysql/dml.go:164]: and some comments to describe the input & output ?

File: cdc/sinkv2/eventsink/txn/mysql/dml.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/parser/charset"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/quotes"
+)
+
```
Comment (nongfushanquan) [inline cdc/sinkv2/eventsink/txn/mysql/dml.go:29]: add some comments to describe the input & output?

**Review by nongfushanquan — APPROVED** [review]

File: cdc/sinkv2/eventsink/txn/mysql/dml.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+import (
+	"context"
+	"database/sql"
+	"strings"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/parser/charset"
+	"go.uber.org/zap"
+
```
Comment (hi-rustin) [inline cdc/sinkv2/eventsink/txn/mysql/dml.go:24]:
```suggestion
```

File: cdc/sinkv2/eventsink/txn/mysql/dml.go [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+		if i > 0 {
+			builder.WriteString(",")
+		}
+		builder.WriteString("?")
+	}
+	return builder.String()
+}
+
+// GetDBConnImpl is the implement holder to get db connection. Export it for tests
+var GetDBConnImpl = getDBConn
```
Comment (hi-rustin) [inline cdc/sinkv2/eventsink/txn/mysql/dml.go:239]: I think we can make this an argument instead of a global variable.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**